core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/typefunction/ToArrayExpression.kt

```diff
@@ -0,0 +1,38 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.typefunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.AtomType
+import ch.ergon.dope.validtype.ValidType
+
+class ToArrayExpression<T : AtomType>(
+    private val expression: TypeExpression<T>,
+) : TypeExpression<ArrayType<T>>, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
+        val expressionDopeQuery = expression.toDopeQuery()
+        return DopeQuery(
+            queryString = toFunctionQueryString("TOARRAY", expressionDopeQuery),
+            parameters = expressionDopeQuery.parameters,
+        )
+    }
+}
+
+@JvmName("atomToArray")
+fun <T : AtomType> toArray(expression: TypeExpression<T>) = ToArrayExpression(expression)
+
+class ArrayToArrayExpression<T : ValidType>(
+    private val expression: TypeExpression<ArrayType<T>>,
```

Comment:
I very much doubt whether we want to offer this functionality in dope: When do we want to pass an array to the toArray-method? Is there a use case?

Reply:
I can remove it no problem. We can always add it again if need be.